### DISCUSSION TITLE
How to create a non-computational visual domain data

### DISCUSSION THREAD

**Discussion by LeighSu** [yoshitomo-matsubara/torchdistill#370] [Q&A] [answered]
Hi Yoshitomo,
 This is an excellent tool.  And Thanks for your shareing.
 I am planning to use this toolkit run KD on  voice-related tasks, suck like keyword spotting, voiceprint recognition.
 I can find the self define loss and model example from other discussion, but I am not sure how to create a new data set, In particular, it's not compatible with torchvision. 
I have found a interest annonation, @register_dataset, but I can not find any example or reference about it?

**Comment by yoshitomo-matsubara** [comment, marked as answer]
Hi @LeighSu 

If you're using torchdistill=v0.3.3 (which I'd suggest at this time),  you can register your own dataset like this

```python
from torchdistill.datasets.registry import register_dataset


@register_dataset
class YourOwnDataset(...):
    ...

```

and specify the dataset name and kwargs in your yaml file
